Backend/controllers/PatientsController.js

```diff
@@ -100,19 +101,73 @@ exports.post_patient_data = async (req, res) => {
     patient._id= mongoose.Types.ObjectId();
 
     try {
-        await patient.save();
-        return res.send(patient);
+        return link_patient_with_user(patient, req, res) 
     } catch (err) {
         return res.status(500).send(err.toString());
     }
 };
 
+// link user with patient...
+link_patient_with_user = async (patient, req, res) =>{
+    let filter;
+    if (patient.email && patient.phone_number){
+        filter = {$or:[{email:patient.email},{phone_number:patient.phone_number}]};
+    } else if (patient.email){
+        filter = {email:patient.email};
+    } else if (patient.phone_number){
+        filter = {phone_number:patient.phone_number}
+    } else {
+        filter = {_id: null} //returns null on User
+    }
+    let user = await User.findOne(filter);
+    if(!user){
+        let unusedUsername = `${patient.first_name}-${patient.last_name}-${Math.random().toString().substring(2, 7)}`;
```

Comment:
I think a substring of Date.now() might help with this random user name generation

Reply:
I see, yeah sure that could work.